Mathlib/Data/Nat/GCD/Basic.lean

```diff
@@ -88,6 +89,33 @@ theorem gcd_self_add_right (m n : ℕ) : gcd m (m + n) = gcd m n := by
   rw [add_comm, gcd_add_self_right]
 #align nat.gcd_self_add_right Nat.gcd_self_add_right
 
+-- Lemmas where one argument consists of a subtraction of the other
+
+theorem gcd_self_sub_symm (m n : ℕ) (h : m ≤ n) : gcd (n - m) n = gcd (n - m) m := by
+  nth_rw 2 [← Nat.add_sub_cancel' h]
+  rw [gcd_add_self_right, gcd_comm]
+
+theorem gcd_sub_self_symm (m n : ℕ) (h : m ≤ n) : gcd n (n - m) = gcd m (n - m) := by
+  rw [gcd_comm, gcd_self_sub_symm m n h, gcd_comm]
+
+@[simp]
+theorem gcd_sub_self_left (m n : ℕ) (h : m ≤ n) : gcd (n - m) n = gcd n m := by
```

Comment:
I wonder if you can prove this more directly without the first two lemmas (which seem of limited utility once the others are established).
Via `gcd n m = gcd (n - m + m) m = gcd (n - m) m`. This should be only two rewrites it seems (sorry I'm on a train and don't have lean available)

Reply:
Yes, I agree that the `...symm` lemmas are not super useful on their own.

Now you mention it: 
The lemma you propose is different still from the lemmas I prove, but probably also useful.
So for the add version we have 4 theorems for all the combinations, but for the sub part we have eualities involving `gcd (n - m) n` and `gcd (n - m) m`, which are different. This leads to 8 theorems.

I don't know what good names are, but I tried to keep it analogous to the corresponding `add`  names.

Reply:
I'm happy with the naming now...